src/actions/moveExtensionDataPath.ts

```diff
@@ -0,0 +1,54 @@
+import * as fs from "fs-extra";
+import * as path from "path";
+import { Result } from "ts-results";
+import * as vscode from "vscode";
+
+import { ExerciseStatus } from "../api/workspaceManager";
+
+import { refreshLocalExercises } from "./refreshLocalExercises";
+import { ActionContext } from "./types";
+
+/**
+ * Moves physical location of all exercises on disk. Closes active course workspace's exercises for
+ * the duration of transfer.
+ *
+ * @param newPath New disk location for exercises.
+ */
+export async function moveExtensionDataPath(
+    actionContext: ActionContext,
+    newPath: vscode.Uri,
+    onUpdate?: (value: { percent: number; message?: string }) => void,
+): Promise<Result<void, Error>> {
+    const { resources, tmc, workspaceManager } = actionContext;
+
+    const activeCourse = workspaceManager.activeCourse;
+    if (activeCourse) {
+        const exercisesToClose = workspaceManager
+            .getExercisesByCourseSlug(activeCourse)
+            .filter((x) => x.status === ExerciseStatus.Open)
+            .map((x) => x.exerciseSlug);
+
+        // Close exercises without writing the result to "reopen" them with refreshLocalExercises
```

Comment:
Should wait for eventListener before opening, maybe.

Reply:
The move may actually work without running that piece of code. Worth a try.